### PR TITLE
meta checks: aggregating multiple checks under a single alias

### PR DESCRIPTION
**Pull request by shlomi-noach**
This PR introduces meta-checks. From the outside a meta check looks just like any other check. But under the cover a meta check is an aggregation of multiple other checks.

For example, the check `api/check/theapp/mysql/cluster1-full` can aggregate multiple checks such as `api/check/theapp/mysql/cluster1-lag` and `api/check/theapp/mysql/cluster1-writeload`.

Checks can have any name. The above check doesn't have to be named `mysql`. It can also be named `api/check/theapp/foo/bar`.

Example configuration:

```json
  "Stores": {
    "Meta": {
        "mysql/cluster1-full": [
            "mysql/cluster1-lag",
            "mysql/cluster1-writeload"
        ]
    },
    "MySQL": {
        ...
    }
```

cc @github/database-infrastructure @miguelff

### CODE REVIEW

File: go/throttle/check.go [hunk 1]
```diff
@@ -58,8 +59,72 @@ func (check *ThrottlerCheck) checkAppMetricResult(appName string, metricResultFu
...
+		}
+	}
+	for _, checkResult = range checkResults {
+		// Now return any good result
+		return checkResult
+	}
+	return NoSuchMetricCheckResult
+}
+
+func (check *ThrottlerCheck) reportMetrics(appName, storeType, storeName, remoteAddr string, statusCode int) {
```
Comment (miguelff) [inline go/throttle/check.go]: Nice extraction.

File: go/throttle/check_test.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+	"github.com/outbrain/golib/log"
+	test "github.com/outbrain/golib/tests"
+)
+
+func init() {
+	log.SetLevel(log.ERROR)
+}
+
+func TestAggregateCheckResults(t *testing.T) {
```
Comment (miguelff) [inline go/throttle/check_test.go]: 👍

**Comment by timvaillancourt** [comment]
@shlomi-noach 👋 this is a really useful PR, thanks!

Merge conflicts aside, is there anything holding this back? I'm happy to get it mergeable and tested

**Comment by shlomi-noach** [comment]
@timvaillancourt if I remember correctly, we ended up not needing these changes, an we opted to not merge the PR so as not to introduce unnecessary code and configuration complications. Again if I remember correctly production deployment looked good with no issues.

**Comment by shlomi-noach** [comment]
Doing personal cleanup of old PRs, this has been stale for 5 years, closing.

**Pull request closed without merging**